From 9cd5184481c4b2de13ba7c70d9a35eac0a994e55 Mon Sep 17 00:00:00 2001
From: "W. Michael Petullo" <mike@flyn.org>
Date: Tue, 24 Mar 2026 09:03:14 -0500
Subject: [PATCH] Fix build with Go 1.26 on Fedora 44 Beta

Fix a handfull of issues related to printf-style functions and format
string verb mismatches.

Signed-off-by: W. Michael Petullo <mike@flyn.org>
---
 commands/hugobuilder.go                        | 8 ++++----
 common/hstore/scratch_test.go                  | 4 ++--
 common/loggers/logger.go                       | 8 ++++----
 transform/livereloadinject/livereloadinject.go | 2 +-
 4 files changed, 11 insertions(+), 11 deletions(-)

diff --git a/commands/hugobuilder.go b/commands/hugobuilder.go
index 4037f6611..c8aaab118 100644
--- a/commands/hugobuilder.go
+++ b/commands/hugobuilder.go
@@ -195,12 +195,12 @@ func (c *hugoBuilder) initMemProfile() {
 
 	f, err := os.Create(c.r.memprofile)
 	if err != nil {
-		c.r.logger.Errorf("could not create memory profile: ", err)
+		c.r.logger.Errorf("could not create memory profile: %s", err)
 	}
 	defer f.Close()
 	runtime.GC() // get up-to-date statistics
 	if err := pprof.WriteHeapProfile(f); err != nil {
-		c.r.logger.Errorf("could not write memory profile: ", err)
+		c.r.logger.Errorf("could not write memory profile: %s", err)
 	}
 }
 
@@ -367,7 +367,7 @@ func (c *hugoBuilder) newWatcher(pollIntervalStr string, dirList ...string) (*wa
 			case changes := <-c.r.changesFromBuild:
 				unlock, err := h.LockBuild()
 				if err != nil {
-					c.r.logger.Errorln("Failed to acquire a build lock: %s", err)
+					c.r.logger.Errorf("Failed to acquire a build lock: %s\n", err)
 					return
 				}
 				c.changeDetector.PrepareNew()
@@ -387,7 +387,7 @@ func (c *hugoBuilder) newWatcher(pollIntervalStr string, dirList ...string) (*wa
 			case evs := <-watcher.Events:
 				unlock, err := h.LockBuild()
 				if err != nil {
-					c.r.logger.Errorln("Failed to acquire a build lock: %s", err)
+					c.r.logger.Errorf("Failed to acquire a build lock: %s\n", err)
 					return
 				}
 				c.handleEvents(watcher, staticSyncer, evs, configSet)
diff --git a/common/hstore/scratch_test.go b/common/hstore/scratch_test.go
index 1f08d46a9..741826caf 100644
--- a/common/hstore/scratch_test.go
+++ b/common/hstore/scratch_test.go
@@ -70,7 +70,7 @@ func TestScratchAddSlice(t *testing.T) {
 	expected := []int{1, 2, 3}
 
 	if !reflect.DeepEqual(expected, sl) {
-		t.Errorf("Slice difference, go %q expected %q", sl, expected)
+		t.Errorf("Slice difference, go %v expected %v", sl, expected)
 	}
 	_, err = scratch.Add("intSlice", []int{4, 5})
 
@@ -80,7 +80,7 @@ func TestScratchAddSlice(t *testing.T) {
 	expected = []int{1, 2, 3, 4, 5}
 
 	if !reflect.DeepEqual(expected, sl) {
-		t.Errorf("Slice difference, go %q expected %q", sl, expected)
+		t.Errorf("Slice difference, go %v expected %v", sl, expected)
 	}
 }
 
diff --git a/common/loggers/logger.go b/common/loggers/logger.go
index ba26137cd..dac9f1a6c 100644
--- a/common/loggers/logger.go
+++ b/common/loggers/logger.go
@@ -237,7 +237,7 @@ func (l *logAdapter) Debugf(format string, v ...any) {
 }
 
 func (l *logAdapter) Debugln(v ...any) {
-	l.debugl.Logf(l.sprint(v...))
+	l.debugl.Logf("%s", l.sprint(v...))
 }
 
 func (l *logAdapter) Info() logg.LevelLogger {
@@ -253,7 +253,7 @@ func (l *logAdapter) Infof(format string, v ...any) {
 }
 
 func (l *logAdapter) Infoln(v ...any) {
-	l.infol.Logf(l.sprint(v...))
+	l.infol.Logf("%s", l.sprint(v...))
 }
 
 func (l *logAdapter) Level() logg.Level {
@@ -318,7 +318,7 @@ func (l *logAdapter) WarnCommand(command string) logg.LevelLogger {
 }
 
 func (l *logAdapter) Warnln(v ...any) {
-	l.warnl.Logf(l.sprint(v...))
+	l.warnl.Logf("%s", l.sprint(v...))
 }
 
 func (l *logAdapter) Error() logg.LevelLogger {
@@ -330,7 +330,7 @@ func (l *logAdapter) Errorf(format string, v ...any) {
 }
 
 func (l *logAdapter) Errorln(v ...any) {
-	l.errorl.Logf(l.sprint(v...))
+	l.errorl.Logf("%s", l.sprint(v...))
 }
 
 func (l *logAdapter) Errors() string {
diff --git a/transform/livereloadinject/livereloadinject.go b/transform/livereloadinject/livereloadinject.go
index 55e69f912..335233a7e 100644
--- a/transform/livereloadinject/livereloadinject.go
+++ b/transform/livereloadinject/livereloadinject.go
@@ -73,7 +73,7 @@ func New(baseURL *url.URL) transform.Transformer {
 		copy(c[idx+len(script):], b[idx:])
 
 		if _, err := ft.To().Write(c); err != nil {
-			loggers.Log().Warnf("Failed to inject LiveReload script:", err)
+			loggers.Log().Warnf("Failed to inject LiveReload script: %s", err)
 		}
 		return nil
 	}
-- 
2.53.0

